init.lua

```diff
@@ -1,11 +1,40 @@
 -- This file includes common functionality that should be available globally
+if Spring.CommonFunctionsInitialized then
+	return
+end
```

Comment:
This works, but I think ideally each thing would protect itself, like

```lua
-- thingFunctions.lua
if Spring.Thing then
    -- already defined!
    return
end
Spring.Thing = ...
```

This way:
 * `VFS.Include("thingFunctions.lua")` is always safe too, as opposed to only `init.lua` (maybe a junior dev sees the file and doesn't realize its contents already available (implicitly included globals are scary!).
 * one less global variable to pollute the `Spring` table and to be aware of (maybe somebody lists all globals as a form of crude API discovery).
 * a bit of a theoretical possibility, but you gain the ability to disable and reenable a single module (unsure why), or to load a module independently earlier than others (for example if you want to poison the `Spring` table for measurements, but need the fancy extra `math` or `table` functions to do it) without doing it a second time in the normal init phase.

Reply:
I changed all the Spring globals to be gated, but the pure Lua function files still get loaded each time. I'm not sure those ones can be gated, since they are extending the base Lua `math`, `string`, and `table` functions - that is why the `Spring.CommonFunctionsInitialized` was left in.

Reply:
All functions now gate themselves